caddyhttp/httpserver/replacer.go

```diff
@@ -141,6 +141,14 @@ func NewReplacer(r *http.Request, rr *ResponseRecorder, emptyValue string) Repla
 		rep.replacements[headerReplacer+strings.ToLower(header)+"}"] = func() string { return strings.Join(values, ",") }
 	}
 
+	// Environment variable placeholders
+	for _, osEnv := range os.Environ() {
+		env := strings.SplitN(osEnv, "=", 2)
+		if len(env) == 2 {
+			rep.replacements[environReplacer+strings.ToLower(env[0])+"}"] = func() string { return env[1] }
+		}
+	}
+
```

Comment:
Environment variables are already evaluated when parsing the Caddyfile. https://caddyserver.com/docs/caddyfile